popup.js

```diff
@@ -322,16 +333,24 @@ function hideNotification() {
     }, 800);
 }
 
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+ }
+
 async function refreshWallet() {
     if (state.walletOpen === true) {
-        let url = getWalletBaseUrl(state.rpcUser, state.rpcPass, state.rpcPort);
+        document.getElementById('load').style.display = "flex";
 
+        let url = getWalletBaseUrl(state.rpcUser, state.rpcPass, state.rpcPort);
         sendCommand(url, "getbalance", [], refreshBalance, (reqStatus, errMessage) => {
             processRequestFail(state.useDebug, reqStatus, errMessage, "refreshWallet getbalance");
         }, state.useDebug);
-        sendCommand(url, "listtransactions", ["*", state.transactionsReturned, 0], processLatestTransactions, (reqStatus, errMessage) => {
+        sendCommand(url, "listtransactions", ["*", state.transactionsReturnedInitially, 0], processLatestTransactions, (reqStatus, errMessage) => {
             processRequestFail(state.useDebug, reqStatus, errMessage, "refreshWallet listtransactions");
         }, state.useDebug);
+
+        await sleep(3000);
+        document.getElementById('load').style.display = "none";

```

Comment:
this should happen in the success/fail methods that follow up "listtransactions"